src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2062,9 +2064,20 @@ namespace winrt::TerminalApp::implementation
     // - <none>
     void TerminalPage::ToggleFocusMode()
     {
-        _isInFocusMode = !_isInFocusMode;
-        _UpdateTabView();
-        _FocusModeChangedHandlers(*this, nullptr);
+        _SetFocusMode(!_isInFocusMode);
+    }
+
+    void TerminalPage::_SetFocusMode(const bool inFocusMode)
+    {
+        // If we're the quake window, we must always be in focus mode.
+        // Prevent leaving focus mode here.
```

Comment:
I know we're on a tight budget and I'd be happy to approve this PR regardless, but technically...

This doesn't strike me as the right kind of abstraction for me. It's like a... inverse leaky abstraction. Basically a leaky injection.
I'd suggest to keep `TerminalPage` unaware about what meaning a window name has, as such logic should be controlled centrally in a single place if possible. Instead a "DisableFocusModeChanges" property might be appropriate.